src/hotspot/share/runtime/init.cpp

```diff
@@ -120,6 +121,8 @@ jint init_globals() {
   compilationPolicy_init();
   codeCache_init();
   VM_Version_init();              // depends on codeCache_init for emitting code
+  // Initialize CPUTimeCounters object, which must be done before creation of the heap.
```

Comment:
Would it be possible to move this inside `universe_init()` in universe.cpp, somewhere above `Universe::initialize_heap()`? There's a similar `MetaspaceCounters::initialize_performance_counters()` in `universe_init()`.